File: backend/src/main/kotlin/de/predic8/openapicatalogbackend/model/entity/document/LintReport.kt
package de.predic8.openapicatalogbackend.model.entity.document

import de.predic8.openapicatalogbackend.model.entity.AEntity
import de.predic8.openapicatalogbackend.model.entity.majorVersion.LintRule
import jakarta.persistence.Entity
import jakarta.persistence.Lob
import jakarta.persistence.ManyToOne

@Entity
class LintReport(@Lob var content: String, @ManyToOne(optional = false) var report: Report, @ManyToOne var rule: LintRule? = null) : AEntity()

File: frontend/src/model/document/LintReport.ts
import AEntity from 'src/model/AEntity'
import type IReportLintEntry from 'src/model/document/IReportLintEntry'
import type LintRule from 'src/model/majorVersion/LintRule'

export default class LintReport extends AEntity {
  content!: IReportLintEntry[]
  rule?: LintRule
}
